configure.ac

```diff
@@ -939,12 +939,24 @@ if test "$use_hardening" != "no"; then
   dnl When enable_debug is yes, all optimizations are disabled.
   dnl However, FORTIFY_SOURCE requires that there is some level of optimization, otherwise it does nothing and just creates a compiler warning.
   dnl Since FORTIFY_SOURCE is a no-op without optimizations, do not enable it when enable_debug is yes.
+  dnl FORTIFY_SOURCE is also disabled for msan/asan/tsan builds since they do not work well together.
   if test "$enable_debug" != "yes"; then
     AX_CHECK_PREPROC_FLAG([-D_FORTIFY_SOURCE=2],[
       AX_CHECK_PREPROC_FLAG([-U_FORTIFY_SOURCE],[
         HARDENED_CPPFLAGS="$HARDENED_CPPFLAGS -U_FORTIFY_SOURCE"
       ])
-      HARDENED_CPPFLAGS="$HARDENED_CPPFLAGS -D_FORTIFY_SOURCE=2"
+
```

Comment:
I'd prefer skipping this detection when the sanitizers are in use, instead of doing this after the detection, which was pointless.

Reply:
I chose this path since compilers may define it by default and for sanitizers it would have to be explicitly undefined (https://github.com/bitcoin/bitcoin/pull/3242/commits/9b4e03b27b04f6dc687409a13859d59bb7909d8f)

Reply:
Ok, will leave further review of the build system changes to others.

Reply:
> I chose this path since compilers may define it by default and for sanitizers it would have to be explicitly undefined 

I don't see the problem with explicitly undefining `FORTIFY_SOURCE` in the (limited) cases where we know it's not going to work. In general I don't like the idea of entangling logic for disabling hardening features that break external tools, inside the happy path of having hardening features used by default. It would be nicer if this existed in one of the [`use_sanitizers` blocks](https://github.com/bitcoin/bitcoin/blob/094d9fda5ccee7d78a2e3d8b1eec17b8b6a33466/configure.ac#L392).

Reply:
I will look into making this change